k8s tenant cleaner job

jbtrystram opened the pull request:
### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

jbtrystram committed:
draft

ctron commented on pkg/controller/iotproject/device_registry_tenant_cleaner.go:
```diff
@@ -0,0 +1,67 @@
+func createIotTenantCleanerJob(name string, iotconfig iotconfig) batchv1.Job {
+
+	job := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{Name: "iot-registry-tenant-cleaner"},
```
This would be unique, and so we need another way of naming. Otherwise we could have only one cleaner. However it should also not be random, as that might result in multiple cleanup jobs for one tenant … maybe use the unique tenant + timestamp … in form that kubernetes allows it.

ctron commented on pkg/controller/iotproject/finalizers.go:
```diff
@@ -139,3 +143,18 @@ func cleanupResource(ctx context.Context, c client.Client, project *iotv1alpha1.
+
+	job := createIotTenantCleanerJob(project.Name)
+
+	result1, err1 := jobsClient.Create(job)
```
You need to gracefully handle the fact that the job exists. Also if the job exists, you need to check the status of the job, and let the finalizer be removed if the job succeeded. Otherwise still wait.

When creating a job fails, that would be an error however. Unless it failed to due to "already exists" reasons.

jbtrystram committed:
i have no idea what i am doing

k-wall committed:
Workaround Fabric8 NPE that may erronously occur whilst awaiting pods (#3400)

* Workaround Fabric8 NPE that may erronously occur whilst awaiting pods

k-wall committed:
Use SASL plain for the Serviceaccount authentication type (#3399)

* Use SASL plain for the Serviceaccount authentication type

The application's AMQP client must be configured to use SASL plain when using this authentication type.

* Update documentation/modules/con-user-model.adoc

Commit:
Add missing env overrides for new images

famarting committed:
improve upgrade test for slow and resource constrained environments (#3402)

k-wall committed:
Resolve router deprecation warning logged during autolink creation (#3319)

"The 'addr' attribute of autoLink has been deprecated. Use 'address' instead"

jenmalloy committed:
Docs issue-3404: fix sizing guide formatting and link issues (#3405)

* Docs issue-3404: fix sizing guide formatting and link issues

* address SME review comment

Commit:
Add task template

Commit:
Further issue template updates

Commit:
Add more guidelines on labels

Commit:
Set fixed minikube version

Commit:
Merge pull request #3409 from lulf/fix-missing-role-permission

Fix missing role permission for configmaps

Commit:
Use non-visible comments for PR template (#3408)

zschwarz committed:
Soak suite cleanup, add load test, add create address spaces method (#3412)

* Soak suite cleanup, add load test, add create address spaces method

obabec committed:
Network policy tests (#3337)

* Network policy tests

dependabot[bot] committed:
Bump jackson from 2.9.10 to 2.10.0 (#3414)

obabec committed:
Create large messages tests (#3278)

obabec committed:
Fix web console tests (#3415)

k-wall committed:
Bump fabric8 kubernetes-client from 4.1.3 to 4.6.2 (#3076)

* Bump fabric8 kubernetes-client from 4.1.3 to 4.6.2

* Bump sundrio.version from 0.16.4 to 0.17.2
* Remove KubernetesDeserializer thread safety workaround.
* Simplify code used to turn off HTTP/2, utilizing new method in KC 4.4.0
* Remove generic parameter from KubernetesResource (relates to https://github.com/fabric8io/kubernetes-client/issues/1661)
* Workaround Fabric8 API change by making cascade desire explict (see https://github.com/fabric8io/kubernetes-client/issues/1840)
